new-components/reporters/pdf/internal/reporter/reporter.go

```diff
@@ -0,0 +1,204 @@
+	if err != nil {
+		slog.Error("could not launch playwright: %s", slog.String("err", err.Error()))
+	}
+
+	defer func() {
+		if err := pw.Stop(); err != nil {
+			slog.Error("could not stop Playwright", slog.String("err", err.Error()))
+		}
+	}()
+
+	slog.Info("reading PDF")
+	resultFilename, pdfBytes, err := p.buildPdf(findings, pw)
+	if err != nil {
+		return "", nil, fmt.Errorf("could not build pdf: %w", err)
+	}
+	slog.Info("result filename", slog.String("filename", resultFilename))
+
+	return resultFilename, pdfBytes, nil
+}
+
+//go:embed template.html
+var templateFile string
+
+// buildPdf builds a PDF
+func (p PdfReporter) buildPdf(data any, pw playwright.Wrapper) (string, []byte, error) {
+	// process the default template into a html result
+	tmpl, err := template.New("template.html").Funcs(template.FuncMap{
+		"formatTime": FormatTime,
+	}).Parse(templateFile)
+	if err != nil {
+		return "", nil, fmt.Errorf("could not parse files: %w", err)
+	}
+
+	currentPath, err := os.Getwd()
+	if err != nil {
+		return "", nil, fmt.Errorf("could not get current working directory: %w", err)
+	}
+
+	reportHTMLPath := filepath.Join(currentPath, "report.html")
+	//#nosec: G304
+	f, err := os.OpenFile(reportHTMLPath, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0o600) //#nosec: G304
+	if err != nil {
+		return "", nil, fmt.Errorf("could not open report.html: %w", err)
+	}
+	if err = tmpl.Execute(f, data); err != nil {
+		return "", nil, fmt.Errorf("could not apply data to template: %w", err)
+	}
+	// close the file after writing it
+	defer func(f *os.File) {
+		err := f.Close()
+		if err != nil {
+			slog.Error("could not close file", slog.String("err", err.Error()))
+		}
+	}(f)
+
+	//todo: add instance id to name of file
+	reportPDFPath := filepath.Join(currentPath, "report.pdf")
+	reportPage := fmt.Sprintf("file:///%s", reportHTMLPath)
+	pdfBytes, err := pw.GetPDFOfPage(reportPage, reportPDFPath)
+	if err != nil {
+		return "", nil, fmt.Errorf("could not generate pdf from page %s, err: %w", reportPage, err)
+
+	}
+
+	// delete the intermediate HTML file
+	if err := os.Remove(reportHTMLPath); err != nil {
+		slog.Error("could not delete report.html", slog.String("err", err.Error()))
+	}
+	return reportPDFPath, pdfBytes, err
+}
+
+// FormatTime is a template function for the PDF, that converts a timestamp to a human-readable format
+func FormatTime(timestamp *int64) string {
+	if timestamp == nil {
+		return ""
+	}
+
+	// Convert the int64 value to a time.Time
+	parsedTime := time.Unix(*timestamp, 0)
+
+	// Format the time using a predefined layout
+	return parsedTime.Format(time.DateTime)
+}
+
+// uploadToS3 uploads the PDF to AWS
+func (p PdfReporter) uploadToS3(resultFilename string, pdfBytes []byte) error {
+	if p.conf.Bucket == "" {
+		slog.Error("bucket is empty, you need to provide a bucket name")
+	}
+
+	if p.conf.Region == "" {
+		slog.Error("region is empty, you need to provide a region name")
+	}
+	client, err := s3client.NewClient(p.conf.Region)
```

Comment:
I'd initialise this too in the main so we can fail early in case the feature is enabled :)